### DISCUSSION TITLE
Problem in adding successfully the DeDRM plugin in calibre

### DISCUSSION THREAD

**Discussion by madldimou** [apprenticeharper/DeDRM_tools#2000] [Q&A]
I do all the steps correctly, but everytime I add the plugin in calibre and I restart it the plugin isn't there.

**Comment by ElleKayEm** [comment]
You're unzipping the main file and loading DeDRM_plugin.zip into calibre?

**Reply by madldimou** [reply]
yes exactly

**Reply by ElleKayEm** [reply]
Preferences > plugins > load plugin from file.  Click ok and apply on your way out of the dialog boxes.  Then restart calibre.

**Reply by madldimou** [reply]
yes I tried it many times and it doesn't work

**Reply by ElleKayEm** [reply]
What version of calibre and what version of DeDRM?

**Reply by madldimou** [reply]
calibre 5.35 and I tried it with dedrm 10.0.2 , 7.2.1  and 7.2.0

**Reply by ElleKayEm** [reply]
I don't know.  You don't want to use 7.2.0 because it has an error, but any of them should install.

**Reply by ElleKayEm** [reply]
You're looking for it under file type plugins?

**Reply by madldimou** [reply]
Yes I do all the steps, I press apply and when I restart the app, the plugin isn't there.

**Reply by ElleKayEm** [reply]
I'm mystified.  I can't duplicate the problem.  It works every time for me.  So it's in the list of plugins when you first load it, but gone after you restart calibre?

**Reply by madldimou** [reply]
Yes that is the issue

**Reply by ElleKayEm** [reply]
If you are looking in Preferences > Get plugins to enhance calibre > Installed, it won't be listed there.

**Reply by madldimou** [reply]
No the proof that the plugin isn't there is that the file does nos get decoded

**Reply by ElleKayEm** [reply]
That doesn't mean it isn't there.  What type of file are you trying to decrypt and how are you downloading it?

**Reply by madldimou** [reply]
It is a book that is decrypted and only opens with adobe digital.

**Reply by ElleKayEm** [reply]
What version of Adobe Digital Editions?  Is it installed on the same machine and user as calibre with DeDRM?  Is ADE authorized with your Adobe ID?

**Reply by madldimou** [reply]
Adobe digital editions 4.5 and yes it is installed in the same device and user. Ade is authorized because otherwise I could not open the book with adobe

**Reply by ElleKayEm** [reply]
You could open the book, but it would be tied to that installation.  With ADE 4.5, you may have hardened Adobe DRM which cannot be decrypted.  Let's see a debug log:  https://github.com/noDRM/DeDRM_tools/blob/master/FAQs.md#i-cannot-solve-my-problem-with-the-dedrm-plugin-and-now-i-need-to-post-a-log-how-do-i-do-that

**Reply by madldimou** [reply]
I did it and here is the log
calibre 5.35 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.22000 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.22000')
Python 3.8.5
Windows: ('10', '10.0.22000', '', 'Multiprocessor Free')
Interface language: el
Successfully initialized third party plugins: DeDRM (10, 0, 2)
calibre 5.35 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.22000 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.22000')
Python 3.8.5
Windows: ('10', '10.0.22000', '', 'Multiprocessor Free')
Interface language: el
Successfully initialized third party plugins: DeDRM (10, 0, 2)
Turning on automatic hidpi scaling
devicePixelRatio: 1.0
logicalDpi: 120.0 x 120.0
physicalDpi: 141.7674418604651 x 141.4020618556701
Using calibre Qt style: True
[0.00] Starting up...
[0.00] Showing splash screen...
[0.89] splash screen shown
[0.89] Initializing db...
[0.91] db initialized
[0.91] Constructing main UI...
[1.95] main UI initialized...
[1.95] Hiding splash screen
Starting QuickView
[24.36] splash screen hidden
[24.36] Started up in 24.36 seconds with 0 books
stdout+stderr from file dialog helper: [b'', b'']
piped data from file dialog helper: [b'\xd3\xb7\xf1r\x1c\xc2\xc3\xc0\xe7\xc0\xf9\x16\x1c\xbe;\xc2\x8a\xc3?L6*\xde\xf6\x06=\xdf\xf8,\xban\xf2', b'C:\\Users\\okapa\\Downloads\\URLLink.acsm']
Added URLLink to db in: 0.2
Added 1 books in 0.6 seconds

**Reply by ElleKayEm** [reply]
You haven't added the actual ebook file here.  Just the acsm file.

**Reply by madldimou** [reply]
All I have is that file.

**Reply by madldimou** [reply]
I mean I bought the book like that and because it is encrypted I am not able to transform it.

**Reply by ElleKayEm** [reply]
When you open an acsm file in ADE, the epub or pdf file is downloaded to your Documents/My Digital Editions folder.

**Reply by madldimou** [reply]
That fixed it! Thank you so much for your help!

**Reply by Eirobal** [reply]
I have the same issue. I unzip the DeDRM_tools-7.2.1 file. The extracted file does not have a DeDRM_plugin.zip file. It has a DeDRM_plugin folder that won't load in Calibre log plugin from file...
I zipped the DRM_plugin folder and it also does not load on the Calibre load plugin utility. 
Please help

**Reply by ElleKayEm** [reply]
Download the first file under Assets here:  https://github.com/apprenticeharper/DeDRM_tools/releases/tag/v7.2.1

Make sure your browser doesn't automatically unzip everything.  Then you should find the DeDRM_plugin.zip file to load into calibre.

There's actually a more recent version here:  https://github.com/noDRM/DeDRM_tools/releases/tag/v10.0.2

**Reply by Eirobal** [reply]
Thanks a lot. It works now and I was able to load the plugin!

**Reply by kentak2** [reply]
I have the same type of problem.  Everything was working fine until yesterday when DeDrm seems to have disappeared and I get the notice that the book file is encrypted.  I tried to reload Calibre to 5.43 and used v 7.2.1.  I select  "preferences" "plug ins", "load plug in from file". " I choose DeDrm_plugin.zip"  I click "OK".  Then nothing happens.  I click "apply" --nothing.  What am I doing wrong? Please help. Ken

**Reply by ElleKayEm** [reply]
What type of book are you trying to decrypt?  How are you downloading it?

**Reply by kentak2** [reply]
They are epub books.  The preferences do not seem to load DeDrm files at all in Calibre.

**Reply by ElleKayEm** [reply]
Is DeDRM not in the list?  Did you accidentally disable it?  Or are you maybe running into epubs with hardened Adobe DRM?

**Reply by kentak2** [reply]
There is nothing in the plug-ins list.  When trying to pick the plug-in from file, I can select the zip file and select Okay, but nothing form that point happens.  No questions asking Am I sure?

**Reply by kentak2** [reply]
This happened once about 6 months ago.  I solved it by going back to an earlier version of Calibre-I don't remember the version of Calibre, but it solved the problem. Must I load an earlier version of Calibre?  And which version?

**Reply by ElleKayEm** [reply]
No idea.  I'm using the current calibre 5.43 and DeDRM 10.0.2 with no issues.  I've never seen DeDRM just disappear from calibre in all the years I've been using it.  If you're on Windows, do you have calibre 32-bit and 64-bit or calibre portable installed at the same time?

**Reply by kentak2** [reply]
I'm on Windows 10 64 bit version of Calibre.

**Reply by ElleKayEm** [reply]
That's what I use as well.  You don't have 32-bit or portable installed too?  I have no idea why it won't work for you.

**Reply by kentak2** [reply]
No. It's juts the 64 bit version.
